porcupine/plugins/filemanager.py

```diff
@@ -122,6 +156,38 @@ def rename(old_path: Path) -> None:
         tab.path = new_path / tab.path.relative_to(old_path)
 
 
+def paste(new_path: Path) -> None:
+    paste_here(new_path.parent)
+
+
+def paste_here(new_path: Path) -> None:
+    assert copy_path is not None and copy_path.is_file()
+    assert new_path.is_dir()
+
+    new_file_path = new_path / copy_path.name
+
+    if new_file_path.exists():
+        show_overwriting_option = True
+        if new_file_path.parent == copy_path.parent:
+            show_overwriting_option = False
+        path = ask_file_name(new_file_path, is_paste=True, show_overwriting_option=show_overwriting_option)
+        if path is None:
+            return
+        new_file_path = path
+
+    if copy_path != new_file_path:
+        shutil.copy(copy_path, new_file_path)
+        get_directory_tree().refresh()
+        if copy_path == new_file_path:  # if overwrite and file is open refresh page
+            close_tabs(find_tabs_by_parent_path(new_file_path))
+            get_tab_manager().open_file(new_file_path)
```

Comment:
These 3 lines can be deleted. The paths will never be the same, since all this is inside an `if` statement that makes it run only when the paths differ. With copying, it's fine if the original file remains open, and the user has to open the newly created copy explicitly.

Reply:
Yes, here I actually made a conceptual error. Before the last commit this check was `if copy_path.name == new_file_path.name`. Then I changed it like this without thinking more.
The purpose I want to reach with these actions is:
check if the operation is an overwrite -> if `the overwritten file is open in the editor` then `refresh the tab of the file since its content changed`. It is not related to the first copied file. I'll think on how to implement it or I'll implement your solution.

Reply:
Actually it was working when the condition was `if copy_path.name == new_file_path.name`. The problem that made me erroneously changed it was that it did this operation even in the case the copied file was pasted without problems with its name in another dir. 
So I have to check if the names are the same and if there was a conflict before (or you think there's another way we can talk about it).

Reply:
Ah, overwriting an open file is something I didn't think of. We can fix this in a separate PR.

Reply:
This PR had so many hidden edge cases. Hopefully we solved (almost) all of them :)

Reply:
Yeah, I didn't realize it was going to be this tricky either.